docs/root/intro/arch_overview/advanced/data_sharing_between_filters.rst

```diff
@@ -88,3 +89,24 @@ the class definition (e.g., HTTP protocol, requested server name, etc.). In
 addition, it provides a facility to store typed objects in a map
 (``map<string, FilterState::Object>``). The state stored per filter can be
 either write-once (immutable), or write-many (mutable).
+
+.. _arch_overview_advanced_filter_state_sharing:
+
+Filter state sharing
+--------------------
+
+Filter state objects are bound to the lifespan of the associated parent stream.
+However, by marking a downstream object as shared with the upstream connection
+during creation, the object is shared with the upstream connection filter
+state, and its lifespan is extended beyond the original stream. Any upstream
+TCP or HTTP filter can access the shared object. Upstream transport sockets can
+also read the shared objects and customize the creation of the upstream
+transport. For example, the :ref:`internal upstream transport socket
+<envoy_v3_api_msg_extensions.transport_sockets.internal_upstream.v3.InternalUpstreamTransport>`
+copies the shared objects to the internal connection downstream filter state.
+
+The filter state objects that are shared with the upstream also affect the
```

Comment:
Can you provide concrete examples of when we add shared hashable objects? Also is this per hashable obj in the FilterState or hashing the collection?